unittest/client/src/ilm_common_unittests.cpp

```diff
@@ -0,0 +1,350 @@
+        RESET_FAKE(ilmControl_registerShutdownNotification);
+    }
+
+    void TearDown()
+    {
+        if(ilm_isInitialized())
+        {
+            ilm_destroy();
+        }
+    }
+
+    void mock_ilmInitSuccess()
+    {
+        mpp_wlDisplays[0] = (struct wl_display*)&m_wlDisplayFakePointer;
+        SET_RETURN_SEQ(wl_display_connect, mpp_wlDisplays, MAX_NUMBER);
+
+        mp_ilmErrorType[0] = ILM_SUCCESS;
+        SET_RETURN_SEQ(ilmControl_init, mp_ilmErrorType, MAX_NUMBER);
+    }
+
+    static constexpr uint8_t MAX_NUMBER = 1;
+    uint8_t m_wlDisplayFakePointer = 0;
+    struct wl_display* mpp_wlDisplays [MAX_NUMBER] = {nullptr};
+    ilmErrorTypes mp_ilmErrorType[MAX_NUMBER] = {ILM_FAILED};
+};
+
+TEST_F(IlmCommonTest, ilm_init_cannotGetDisplay)
+{
+    // Prepare fake for wl_display_connect, to make the result is nullptr
+    mpp_wlDisplays[0] = nullptr;
+    SET_RETURN_SEQ(wl_display_connect, mpp_wlDisplays, MAX_NUMBER);
+    // Invoke the ilm_init and expect the result is ILM_FAILED
+    ASSERT_EQ(ILM_FAILED, ilm_init());
+    /* Check the logic
+     * wl_display_connect should call one time and return nullptr 
+     */
+    ASSERT_EQ(wl_display_connect_fake.call_count, 1);
+    ASSERT_EQ(wl_display_connect_fake.return_val_history[0], nullptr);
+}
+
+TEST_F(IlmCommonTest, ilm_init_getFailedOnilmControl_init)
+{
+    // Prepare fake for wl_display_connect, to make return different nullptr
+    mpp_wlDisplays[0] = (struct wl_display*)&m_wlDisplayFakePointer;
+    SET_RETURN_SEQ(wl_display_connect, mpp_wlDisplays, MAX_NUMBER);
+    // Prepare fake for ilmControl_init, to make return ILM_FAILED
+    mp_ilmErrorType[0] = ILM_FAILED;
+    SET_RETURN_SEQ(ilmControl_init, mp_ilmErrorType, MAX_NUMBER);
+    // Invoke the ilmControl_init and expect the result is ILM_FAILED
+    ASSERT_EQ(ILM_FAILED, ilm_init());
+    /* Check the logic
+     * ilmControl_init_fake should call one time and return ILM_FAILED
+     */
+    ASSERT_EQ(ilmControl_init_fake.call_count, 1);
+    ASSERT_EQ(mp_ilmErrorType[0], ilmControl_init_fake.return_val_history[0]);
+}
+
+TEST_F(IlmCommonTest, ilm_init_getSuccess)
+{
+    // Prepare fake for wl_display_connect and ilmControl_init, to make the ilm_init return success
+    mock_ilmInitSuccess();
+    // Invoke the ilm_init and expect the resutl is ILM_SUCCESS
+    ASSERT_EQ(ILM_SUCCESS, ilm_init());
+    /* Check the logic
+     * wl_display_connect should call one time
+     * ilmControl_init_fake should call one time and return ILM_SUCCESS
+     */
+    ASSERT_EQ(wl_display_connect_fake.call_count, 1);
+    ASSERT_EQ(ilmControl_init_fake.call_count, 1);
+    ASSERT_EQ(ILM_SUCCESS, ilmControl_init_fake.return_val_history[0]);
+}
+
+TEST_F(IlmCommonTest, ilm_init_manyTimes)
+{
+    // Prepare fake for wl_display_connect and ilmControl_init, to make the ilm_init return success
+    mock_ilmInitSuccess();
+    // Invoke the ilm_init in first time and expect return ILM_SUCCESS
```

Comment:
I see too many comments in the code. Its really not necessary. e.g. Individual comments in this function can be concisely written as test for multiple ilm_init and ilm_destroy calls.